### PR TITLE
🐛Source Hubspot: Fixed 403 error validation

### PR DESCRIPTION
**Pull request by lazebnyi**
## What
https://github.com/airbytehq/oncall/issues/526 - Source: HubSpot - requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.hubapi.com/automation/v3/workflows?limit=100 (via Sentry) 

When facing forbidden error, the Hubspot source connector just log it. Syncs stop after these errors occur.

## How
Handle error and return empty record if error occur.

## Pre-merge Checklist
<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 1]
```diff
@@ -354,12 +360,14 @@ def read_records(
...
                 properties = self._property_wrapper
                 if properties and properties.too_many_properties:
+                    print("lol1")
                     records, response = self._read_stream_records(
                         stream_slice=stream_slice,
                         stream_state=stream_state,
                         next_page_token=next_page_token,
                     )
                 else:
+                    print("lol2")
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:370]: Same comment

**Review by bazarnov — APPROVED** [review]
Please check the `print()` statements left inside the code.

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 2]
```diff
@@ -354,12 +360,14 @@ def read_records(
 
                 properties = self._property_wrapper
                 if properties and properties.too_many_properties:
+                    print("lol1")
```
[the remark this answers is not in the document]
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py]: updated

**Comment by pedroslopez** [comment]
@lazebnyi looks like we have some failing tests

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 3]
```diff
@@ -259,6 +260,11 @@ def __init__(self, api: API, start_date: Union[str, pendulum.datetime], credenti
         elif creds_title in (OAUTH_CREDENTIALS, PRIVATE_APP_CREDENTIALS):
             self._authenticator = api.get_authenticator()
 
+    def should_retry(self, response: requests.Response) -> bool:
+        if response.status_code == HTTPStatus.FORBIDDEN:
+            setattr(self, "raise_on_http_errors", False)
```
Comment (pedroslopez) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:265]: are we logging tot he user when this happens?

Comment (lazebnyi) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:265]: updated

**Pull request merged**